Some basic cleanup

mfojtik opened the pull request:
- Removed Blaa and Sausage controllers/models
- Added Chassis representation instead of complex type

(Please ignore the rest of the commits, this patch apply on top of https://github.com/tuskar/tuskar/pull/3 )

mfojtik committed:
Removed Blaa and Sausage models/controllers

mfojtik committed:
Added Chassis representation type, some basic refactoring

mfojtik committed:
Added Capacity representation to controller

mfojtik commented:
Update: I added Capacity representation instead of using weird wtype ;-) Also I cleaned up how we construct the Rack response to avoid code duplication,.

mfojtik committed:
More refactoring and added delete_rack()

markmc commented:
I'm fine with merging this as is, but I don't totally follow what you're doing with chassis here

Some curl examples of creating a Rack would be good

mfojtik commented:
@markmc right, I added some examples on bottom, just a quick update:
- Added 'delete_rack()' method and controller action
- Some more refactoring on how I create Rack (added DB transaction and stop using .as_dict() when I can pass Rack representation there ;-)

Examples:

Create a new Rack:

``` bash
curl -X POST -v -d '{"subnet":"192.168.1.0/255","name":"my_rack","capacities":[{"name":"total_cpu","value":"64"},{"name":"total_memory","value":"1024"}],"chassis":{"links":[{"href":"http://ironic.local:3333/v1/chassis/123","rel":"self"}]},"slots":1}' -H 'Accept: application/json' -H 'Content-Type: application/json' http://0.0.0.0:6385/v1/racks
```

Get the list of Racks:

``` bash
curl -X GET -v  -H 'Accept: application/json' -H 'Content-Type: application/json' http://0.0.0.0:6385/v1/racks
```

Get the single Rack:

``` bash
curl -X GET -v  -H 'Accept: application/json' -H 'Content-Type: application/json' http://0.0.0.0:6385/v1/racks/1
```

Delete Rack:

``` bash
curl -X DELETE -v  -H 'Accept: application/json' -H 'Content-Type: application/json' http://0.0.0.0:6385/v1/racks/1
```

markmc commented:
Ok, when you create a rack, you shouldn't be specifying an existing ironic chasis - tuskar is responsible for creating it

mfojtik commented:
@markmc aah OK, so then can just remove:

``` python
chassis_url=new_rack.chassis.links[0].href
```

from 'api.py' in sqlachemy and left if blank for Tuskar to update it. Do you want me to do now, as part of this pull request?

mfojtik committed:
Removed unused as_dict() method from Capacity

mfojtik committed:
Do not store the 'chassis_url' when creating a Rack

mfojtik commented:
@markmc nevermind ;-) already did it :-)

markmc commented on tuskar/db/sqlalchemy/models.py:
```diff
@@ -142,18 +116,9 @@ class Rack(Base):
     chassis_url = Column(Text)
     capacities = relationship("Capacity",
             secondary=Base.metadata.tables['rack_capacities'],
+            cascade="all, delete",
```
Bit nervous about this because we don't generally use sqlalchemy's cascading deletes in openstack ... but I see quantum does use it, so let's do it

mfojtik replied:
Yeah, I also haven't managed to cascade delete the 'capacities' themself, just 'rack_capacities', so I removing them in a DB transaction in 'delete_rack' method...

markmc commented:
Looks like good progress
